src/interface.jl

```diff
@@ -86,7 +86,7 @@ end
 #      should probably provide the world age of the calling code (!= the current world age)
 #      so that querying the cache from, e.g. `cufuncton` is a fully static operation.
 FunctionSpec(f, tt=Tuple{}, kernel=true, name=nothing, world_age=-1%UInt) =
```

Comment:
```suggestion
FunctionSpec(f::Type, tt=Tuple{}, kernel=true, name=nothing, world_age=-1%UInt) =
```

and then create a compatibility shim `FunctionSpec(f, tt, args...) = FunctionSpec(Core.Typeof(f), tt, args...` So that users don't need to change their behaviour.